frontend/src/metabase/embedding/embedding-iframe-sdk-setup/components/SdkIframeEmbedSetupModal.tsx

```diff
@@ -148,7 +159,7 @@ export const SdkIframeEmbedSetupContent = () => {
 
           <SdkIframeGuestEmbedStatusBar />
 
-          {allowPreviewAndNavigation ? (
+          {allowPreviewAndNavigation && !isMissingResource ? (
```

Comment:
I have a hunch this is not needed, the reason is that you already have logic to pass `resourceId: null` in hooks.

I tried removing this and the test still passes.